### ISSUE TITLE
Soot does not find default methods reachable

### ISSUE THREAD

**Issue by kalhauge** [soot-oss/soot#1054]
Consider the example:

```java
public class Main implements Default {
  public static void main(String[] args) {
    new Main().target();
  }
}
interface Default {
  default void target() {
    System.out.println("Hello, World!");
  }
}
```

When trying to find reachable methods from main, soot does not find Default.target. Even though
the program print "Hello, World!", when run.

I followed the same setup as in #1045.

## Versions:
Soot version 3.1.0

```
$ java -version
openjdk version "1.8.0_181"
OpenJDK Runtime Environment (build 1.8.0_181-13)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)
```

**Comment by mbenz89** [comment]
@StevenArzt Any ideas on this? Might it be related to the `default` keyword for interfaces, introduced in Java 8?

**Comment by StevenArzt** [comment]
We would have to investigate what default methods in interfaces look like in the bytecode. I guess they are not normal methods in a class as Soot expects them to be. Has anyone run javap on the example? Or maybe looking at the Jimple code is already enough to see what's going on under the hood.

**Comment by mbenz89** [comment]
fixed in #1221
